config/configloader/config.go

```diff
@@ -458,6 +458,12 @@ func expandStringValues(value interface{}) interface{} {
 			nslice = append(nslice, expandStringValues(vint))
 		}
 		return nslice
+	case map[string]interface{}:
```

Comment:
Why was this case not needed for viper and needed now?

Reply:
I'm not entirely sure since both `map[string]interface{}` and `map[interface{}]interface{}` are used in submaps in viper, but konf signatures opt for `map[string]interface{}` - 
https://github.com/knadh/koanf/blob/master/koanf.go#L319
and https://github.com/knadh/koanf/blob/deea8ad19881d954ecf42e1a6015bce109a29114/maps/maps.go#L240

This may be a latent bug?

Reply:
Not sure, but you feel it is warranted please add a test case that exercises this path.

Reply:
I'm actually seeing the `map[interface{}]interface{}` case is never exercised when using koanf and the added case is exercised by all existing tests.

@pjanotti are you familiar with any potential consequences from doing?

```diff
diff --git a/config/configloader/config.go b/config/configloader/config.go
index c8dc9d5c..40a326ca 100644
--- a/config/configloader/config.go
+++ b/config/configloader/config.go
@@ -459,13 +459,7 @@ func expandStringValues(value interface{}) interface{} {
                }
                return nslice
        case map[string]interface{}:
-               nmap := make(map[interface{}]interface{}, len(v))
-               for k, vint := range v {
-                       nmap[k] = expandStringValues(vint)
-               }
-               return nmap
-       case map[interface{}]interface{}:
-               nmap := make(map[interface{}]interface{}, len(v))
+               nmap := make(map[string]interface{}, len(v))
                for k, vint := range v {
                        nmap[k] = expandStringValues(vint)
                }
diff --git a/config/internal/configsource/manager.go b/config/internal/configsource/manager.go
index ddcf8bf9..6fb9c5dc 100644
--- a/config/internal/configsource/manager.go
+++ b/config/internal/configsource/manager.go
@@ -310,17 +310,7 @@ func (m *Manager) expandStringValues(ctx context.Context, value interface{}) (in
                }
                return nslice, nil
        case map[string]interface{}:
-               nmap := make(map[interface{}]interface{}, len(v))
-               for k, vint := range v {
-                       value, err := m.expandStringValues(ctx, vint)
-                       if err != nil {
-                               return nil, err
-                       }
-                       nmap[k] = value
-               }
-               return nmap, nil
-       case map[interface{}]interface{}:
-               nmap := make(map[interface{}]interface{}, len(v))
+               nmap := make(map[string]interface{}, len(v))
                for k, vint := range v {
                        value, err := m.expandStringValues(ctx, vint)
                        if err != nil {
```

Complex map keys don't appear to be allowable by gopkg.in/yaml.v2 so I don't believe keys will ever be of arbitrary type.  `configparser.Parser.AllKeys()` and `configparser.Parser.ToStringMap()` presume keys will always be strings as well.